src/configuration.ts

```diff
@@ -12,90 +23,158 @@ export interface Config {
+        initial: false,
+        active: "Yes / Ja",
+        inactive: "No / Nein",
+      },
+    ]);
+
+    // Exit when aborted
+    if (!url) {
+      process.exit(1);
+    }
+
+    baseConfiguration.queue.push({
+      url,
+      code,
+    });
+
+    if (!anotherUrl) {
+      break;
+    }
+  }
+
+  debug("Storing your config.json for next time...");
+  fs.writeFileSync(configPaths[0], JSON.stringify(configuration, undefined, 2));
+  debug("Wrote your configuration to %s", configPaths[0]);
+
+  return configuration;
+}
+
+export async function loadConfiguration() {
   let configuration: Config | undefined = undefined;
+
   for (const configPath of configPaths) {
     if (!fs.existsSync(configPath)) continue;
     try {
       configuration = JSON.parse(fs.readFileSync(configPath).toString("utf-8"));
     } catch (e) {
-      debug("There are errors in the config.json %s", e);
+      debug(error("There are errors in the config.json %s", e));
+
+      if (!(await askToCreateNewConfig())) {
+        process.exit(1);
+      }
+
       debug("I am creating a new config.json");
-      configuration = undefined;
+      configuration = await generateNewConfig();
     }
+    break;
   }
+
   if (!configuration) {
-    debug("So, let's get started with your configuration...");
-    debug(
-      "Go to https://github.com/marcoklein/impftermin#getting-an-impfzentrum-url for further information"
-    );
-    debug(
-      "Lies dir die Info in https://github.com/marcoklein/impftermin#getting-an-impfzentrum-url durch"
-    );
-
-    configuration = baseConfiguration;
-    configuration.queue = [];
-
-    while (true) {
-      const { url, anotherUrl, code } = await prompts([
-        {
-          type: "text",
-          name: "url",
-          message:
-            "Enter an Impf location URL you want to check. / Gib die URL des Impfzentrums ein.",
-        },
-        {
-          type: "toggle",
-          name: "codeAvailable",
-          message:
-            "Do you have an Impf code for that location? / Hast du einen Impfcode für dieses Impfzentrum?",
-          initial: false,
-          active: "Yes / Ja",
-          inactive: "No / Nein",
-        },
-        {
-          type: (prev) => (prev ? "text" : null),
-          name: "code",
-          message: "Your Impf code / Dein Impfcode:",
-        },
-        {
-          type: "toggle",
-          name: "anotherUrl",
-          message:
-            "Do you want to search another location? / Möchtest du ein weiteres Impfzentrum hinzufügen?",
-          initial: false,
-          active: "Yes / Ja",
-          inactive: "No / Nein",
-        },
-      ]);
-
-      baseConfiguration.queue.push({
-        url,
-        code,
-      });
-
-      if (!anotherUrl) {
-        break;
+    configuration = await generateNewConfig();
+  }
+
+  try {
+    assert(configuration, Config);
+  } catch (e) {
+    debug(error("Invalid configuration:", e));
+    if (await askToCreateNewConfig) {
```

Comment:
I think this should call the function with `askToCreateNewConfig()`?

Reply:
Yes, of course. I'm surprised I didn't realize this during testing.